src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,183 @@
+import p5 from 'p5' // we need the p5.Color type to declare top-level colors
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# Entries (p5 Template)
+
+[<img
+  src="../../assets/img/Entries/1729.png"
+  width=700
+  style="margin-left: 1em; margin-right: 0.5em"
+/>](../../assets/img/Entries/1729.png)
+
+This visualizer allows you to step back and forth through a sequence, showing
+each entry as a written number.
+
+_This visualizer is meant to be used as a template for new visualizers based on
+the p5.js library. It includes explanatory comments and minimal examples of
+required and commonly used features._
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    // === visualizer name ===
+    // appears in the visualizer list and bundle card titles
+
+    static visualizationName = 'Entries (p5 Template)'
+
+    // === parameters ===
+    // top-level properties that the user can choose while creating the
+    // visualizer bundle
+
+    stepSize = 1 // the default value of this parameter is zero
+
+    /** md
+## Parameters
+    **/
+    params = {
+        /** md
+- **Step size:** How far to step when the user presses an arrow key.
+         **/
+        stepSize: {
+            value: this.stepSize,
+            forceType: 'integer',
+            displayName: 'Step size',
+            required: true,
+        },
+    }
+
+    // === internal properties ===
+    // top-level properties that are set and updated while the visualizer is
+    // running, beyond the user's direct control
+
+    // navigation state. this property will be initialized during setup, but
+    // TypeScript can't infer that, so we have to give a placeholder value
+    index = 0
+
+    // palette colors. these will also be initialized during setup, which is our
+    // first chance to create colors. for now, we use `undefined` as a
+    // placeholder value
+    bgColor: p5.Color | undefined = undefined
+    textColor: p5.Color | undefined = undefined
+
```

Comment:
I personally think it is better to initialize
```
bgColor = new p5.Color()
```
here to avoid essentially turning off TypeScript typechecking by using `as p5.Color` with every use of `this.bgColor` below. It's also all-around more concise.

Reply:
Do you see any downside to initializing in this way?

Reply:
When I tried to initialize using `p5.Color()`, I got runtime errors that looked like "`e4` is not defined", or something. Does it work on your setup?

Reply:
Ugh, I took the p5 documentation at its word, which now says that the p5 instance argument is optional, which looking at its source code most definitely is **not**. This is real wart in p5, that there in fact seems to be no legit way to get a p5.Color instance without having created a sketch.

So here is what I think we should do:
* add a static member to P5Visualizer called WHITE initialized to `{mode: 'rgb', levels: [255,255,255,255], maxes: {rgb: [255,255,255,255]}, _array: [1,1,1,1]} as p5.Color`. I got this by looking at the source and the console. I believe it will be safe to draw with, if it does leak through to a draw call.
* use P5Visualizer.WHITE as the initializer any time we need a color value initialized before any sketch.

Let me know if that sounds reasonable to you and if so whether you want me to do it or to push a commit doing it.

Reply:
Oh, no, it likely won't be safe to draw with, as it won't have the right prototype. Two options:
* There is some JavaScript incantation that will attach the correct prototype to this object, I could figure it out.
* We could instead export P5Visualizer.INVALID_COLOR = `{} as p5.Color` and that should make it clear that you need to replace that with a color from a sketch before using it. In other words, have just one "blessed" instance of this icky hack, that if we find a way to replace in the future if we can, rather than lots of TypeScript gyrations in every single Visualizers that uses colors but needs to initialize some before there is a sketch.

Reply:
I agree that we should get rid of the `p5.Color | undefined` union type if possible; programming around it generates lots of friction and needless complexity. I don't know TypeScript well enough to evaluate the ideas you proposed, though. I think we can safely leave this for a later PR.

Reply:
I just couldn't bear to be checking in a template that had deliberate subversions of the TypeScript type system. (Don't get me wrong, I hate TypeScript, but this is a TypeScript project so we should be true to its nature.) So I "implemented" the simplest INVALID_COLOR for now. If we get to a static working color, it will be easy to switch; the code will be the same except for changing INVALID_COLOR to BLACK, say. Resolving.